docs/source/conf.py

```diff
@@ -77,6 +77,14 @@
 # and we document the types anyway
 autodoc_typehints = "none"
 
+# Show docstring for special members
+autodoc_special_members = ["__aenter__", "__aexit__"]
+
+autodoc_default_options = {
+    "special-members": True,
+    "exclude-members": "__init__",
+}
+
```

Comment:
Nice - does that mean that we can remove lines like 

https://github.com/python-telegram-bot/python-telegram-bot/blob/1ef242a17e9bcb7a9229664795da9677c2eeb72f/docs/source/telegram.telegramobject.rst?plain=1#L7

?

I don't guide unnderstand the first two lines though. That option doesn't seem to exist (I only found mentions in https://github.com/sphinx-doc/sphinx/issues/4075 and https://github.com/sphinx-doc/sphinx/pull/4076, but the latter was closed in favor of https://github.com/sphinx-doc/sphinx/pull/5215) and lines 84 already seems to care of it?

Reply:
I tested locally and I obsevered:

* `autodoc_special_members = ["__aenter__", "__aexit__"]` has no effect - after removing that line, the methods are still listed in the documentation
* `"special-members": True` does indeed include all dunder methods. After reomving the line indicated above, the methods still show in the docs. Since most of our dunder methods already have docstrings, this does sound like a good thing to me.

I double checked and the only public classes with dundermethods without docstrings are Bot (`__eq__` & `__hash__`) and the classes in the `error` module. I would therefore propose:

1. remove lines 80 & 81 and then merge this PR
2. in a follow-up PR (or just in doc-fixes branch):
    1. add docstrings where missing on dunder methods
    2. Remove the `:special-members:` setting from the `*.rst` files

Would that be okay with you? cc @harshil21

Reply:
yep, sounds good. I also observed that `autodoc_special_members` variable has no effect.

Reply:
great! I would be going with the follow up pr also

Reply:
Openend #3926 for this :) @clot27 would you like to address that yourself?

Reply:
yes sure